enhancements/security/security-profiles-operator.md

```diff
@@ -0,0 +1,636 @@
+---
+title: security-profiles-operator
+authors:
+  - "@jaormx"
+  - "@jhrozek"
+  - "@saschagrunert"
+reviewers:
+  - "@ashcrow"
+  - "@mrunalp"
+  - "@sttts"
+  - "@mrunalp"
+  - "@shawn-hurley"
+  - "@dmesser"
+approvers:
+  - TBD
+creation-date: yyyy-mm-dd
+last-updated: yyyy-mm-dd
+status: provisional|implementable|implemented|deferred|rejected|withdrawn|replaced|informational
+see-also:
+  - "/enhancements/this-other-neat-thing.md"
+replaces:
+  - [selinux-operator proposal](https://github.com/openshift/enhancements/pull/327)
+superseded-by:
+  - "/enhancements/our-past-effort.md"
+---
+
+# Security Profiles Operator Integration in OpenShift
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Let's integrate the [Security Profiles Operator](https://github.com/kubernetes-sigs/security-profiles-operator)
+as part of the OpenShift platform in order to allow folks to:
+
+* Easily install security profiles to secure their workloads
+  (both SELinux and Seccomp)
+
+* Manage the lifecycle of the aforementioned profiles (make sure
+  that they get attached to the appropriate container as upgrades
+  happen)
+
+* Enable users to automatically generate profiles for their workloads
+
+* Provide predefined profiles for existing default workloads in OpenShift
+
+
+## Motivation
+
+OpenShift already has several security controls enabled by default. Amongst them
+are the following:
+
+* SELinux to restrict what resources can containers interact with
+
+* Seccomp to restrict the system calls that containers are allowed to do
+
+* Security Context Constraints to control what security knobs and handles are
+  service accounts able to tweak in their workloads.
+
+The default SELinux and Seccomp profiles work great for restricted workloads.
+However, once the workload needs a little more permissions, setting these
+security settings becomes non-trivial. In face of this, developers often
+respond by turning off these controls and running their workloads as privileged
+containers.
+
+The [Security Profiles Operator (SPO)](https://github.com/kubernetes-sigs/security-profiles-operator)
```

Comment:
@stlaz in the initial pitch of this, the idea was to bring it as a Marketplace operator. But it was brought up that having it as a Marketplace Operator first would make it really hard for folks to adopt the security profiles, as it would not be sure that the operator would be there. So... if you try to install an operator that uses a security profile, the operator would be uninstallable unless the SPO would be there. Hence why we switched to proposing this as a CVO operator already.

Reply:
Perhaps we should be talking about having a way for OLM operators to express dependencies on other operators.